nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -216,10 +224,24 @@ in
 
         requires = lib.optionals cfg.database.createLocally [ "postgresql.service" ];
 
+        path = mkIf (cfg.secretFile != null) [ pkgs.jq ];
+
+        # merge the two configs and prevent others from reading the result
+        # if somehow $CREDENTIALS_DIRECTORY is not set we fail
+        preStart = mkIf (cfg.secretFile != null) ''
+          set -u
+          umask 177
+          jq --slurp '.[0] * .[1]' "${configFile}" "$CREDENTIALS_DIRECTORY/secretFile" > "${mergedConfig}"
```

Comment:
FYI, `genJqSecretsReplacementSnippet` does similar things.

Reply:
nice find!
I think in this case though, the script is simple enough to be left as is.

Thanks a lot for doing a thorough review!

Reply:
Thanks!

I slightly prefer to use that util function for better readability and maintenance.

Reply:
While using this function can increase maintainability it makes it harder to run lemmy as  a dynamic user since we can't use `LoadCredential=` since we don't know which files will be included (unless we scan the config and search for includes, but I think this is a bit much just for using that util function).

Reply:
But I should use `lib.replaceShellArg`

Reply:
> ... (using this function) makes it harder to run lemmy as a dynamic user since we can't use `LoadCredential=` since we don't know which files will be included (unless we scan the config and search for includes ...)

`Dynamicuser` can still be used if we let users set `systemd.services.lemmy.serviceConfig.LoadCredential` either directly or indirectly.  Not a good user interface though.

I do not mean that this module must use `utils.genJqSecretsReplacementSnippet`.  Both methods are fine since, other than using `LoadCredential`, there is no convention for handling secrets in NixOS.